File: analysis/plot_benchmarks.py
import os
from plotnine import *
import pandas as pd
import requests
import re


def main():
    for name, benchmark_data_path in load_benchmarks():
        df = pd.read_csv(benchmark_data_path)
        df['Population size'] = pd.Categorical(df['population_size'])
        print(f"Plotting {name}")
        print(df)
        plt = ggplot(df, aes(x="Iterations", y="Markets visited")) + \
            scale_x_log10(name="Iterations per run") +\
            geom_jitter(width=0.5, height=0.2) +\
            geom_smooth(span=0.75)# geom_jitter(height=0.05) +\
        print(plt)
        # run(name, market_data_path, durations_data_path,
        #     save_runs=True, show_route=False, show_plot_runs=False, save_plot_runs=True)


def load_benchmarks():
    base_path = os.path.join("data", "runs")
    dataset_names = list()
    benchmark_data_paths = list()

    for file in os.listdir(base_path):
        stem, ext = os.path.splitext(file)

        dataset_names.append(stem)
        benchmark_data_paths.append(os.path.join(base_path, file))
    return zip(dataset_names, benchmark_data_paths)


if __name__ == '__main__':
    main()

File: analysis/plot_benchmarks.R
library(ggplot2)
library(scales)
library(cowplot)
time <- 0
for (file in list.files("data/runs", full.names = T))
{

    data <- read.csv(file)
    time <- time + sum(data$runtimes)
    mplt <- ggplot(data, aes(x = Iterations, y = Markets.visited, col = as.factor(population_size))) +
        scale_x_log10(name = "Iterations per run") +
        scale_color_discrete(name = "population size") +
        scale_y_continuous(breaks=pretty_breaks()) +
        geom_jitter(width = 0.1, height = 0.2) +
        geom_smooth(span = 0.75) +
        labs(title = "GA solution quality by population size and amount of iterations", y = "markets visited")

    rplt <- ggplot(data, aes(x = Iterations, y = runtimes, col = as.factor(population_size))) +
        scale_x_log10(name = "Iterations per run") +
        scale_y_log10(name = "log runtimes/s", labels = scales::comma) +
        scale_color_discrete(name = "population size") +
        geom_jitter(width = 0.025, height = 0.05) +
        geom_smooth(span = 0.75) +
        labs(title = "GA solution time by population size and amount of iterations")

    plt <- plot_grid(mplt, rplt, nrow = 2)
    sub("[.].*", "", file)
    ggsave(paste0("data/figures/", gsub("data/runs/|_runs.csv", "", file), "_ggplot.png"), plt)

}
print(time)
